### PR TITLE
[CPU] Add `contract` fast-math-flag to arith operations

### PR DESCRIPTION
**Pull request by dcaballe**
This patch adds the `contract` FMF to some arith operations so that they can be folded into an fma instruction. We are doing this by default as we are lowering matmul ops by default to fmas. We will add different fp modes to have more control on fp optimizations depending on the tolerance to fp errors.

### CODE REVIEW

**Comment by MaheshRavishankar** [comment]
Nice! I am not familiar with the contract fast math flag. Do you have a link for reference

**Comment by MaheshRavishankar** [comment]
Nice! I am not familiar with the contract fast math flag. Do you have a link for reference ?

**Comment by dcaballe** [comment]
https://llvm.org/docs/LangRef.html#fast-math-flags

**Comment by MaheshRavishankar** [comment]
So what does it mean to add it fsub, fneg?

File: compiler/src/iree/compiler/Codegen/Common/AddFastMathFlags.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  LLVM::FastmathFlags contract = LLVM::FastmathFlags::contract;
+  TypeSwitch<Operation *>(op)
+      .Case<LLVM::FMulOp>(
+          [&](auto llvmOp) { llvmOp.setFastmathFlags(contract); })
+      .Case<LLVM::FAddOp>(
+          [&](auto llvmOp) { llvmOp.setFastmathFlags(contract); })
+      .Case<LLVM::FSubOp>(
+          [&](auto llvmOp) { llvmOp.setFastmathFlags(contract); })
+      .Case<LLVM::FNegOp>(
+          [&](auto llvmOp) { llvmOp.setFastmathFlags(contract); });
```
Comment (hanhanW) [inline compiler/src/iree/compiler/Codegen/Common/AddFastMathFlags.cpp]: I'm not fully sure, but I think we can merge these cases into a single `Case`. Then we can save 6 lines.

```suggestion
      .Case<LLVM::FMulOp, LLVM::FAddOp, LLVM::FSubOp, LLVM::FNegOp>(
          [&](auto llvmOp) { llvmOp.setFastmathFlags(contract); })
```

**Review by hanhanW — APPROVED** [review]
This is new to me! I have same questions about what's happening to `fsub` and `fneg`.

(I approved the PR because the code itself looks good to me; it is a thing we wanna control at this level)

**Comment by dcaballe** [comment]
> So what does it mean to add it fsub, fneg?

They can be folded into an fmsub instruction :)

**Review by MaheshRavishankar — APPROVED** [review]

**Pull request merged**